pkg/operator/operator.go

```diff
@@ -638,3 +650,34 @@ func RedeployAppVersion(appID string, sequence int64) error {
 
 	return nil
 }
+
+func deduplicateSecrets(secretSpecs []string) []string {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+
+	uniqueSecrets := map[string]*corev1.Secret{}
+	for _, secretSpec := range secretSpecs {
+		obj, gvk, err := decode([]byte(secretSpec), nil, nil)
+		if err != nil {
+			continue
+		}
+
+		if gvk.Group != "" || gvk.Version != "v1" || gvk.Kind != "Secret" {
+			continue
+		}
+		secret := obj.(*corev1.Secret)
+		uniqueSecrets[secret.Name] = secret
```

Comment:
what about namespace?

Reply:
they are applied to all namespaces.  we just need the list of secrets without namespaces.

Reply:
here https://github.com/replicatedhq/kots/blob/847a03667df7c19b59398399f34d7ad67dd8c03c/pkg/operator/client/deploy.go#L267-L268